packages/cli/test/integration/generators/controller.integration.js

```diff
@@ -1,425 +1,318 @@
+const testUtils = require('../../test-utils');
 
-const templateName = testUtils.givenAControllerPath(
-  null,
-  'controller-template.ts',
-);
-const withInputProps = {
+// Test Sandbox
+const SANDBOX_PATH = path.resolve(__dirname, '..', '.sandbox');
+const sandbox = new TestSandbox(SANDBOX_PATH);
+
+// CLI Inputs
+const basicCLIInput = {
+  name: 'productReview',
+};
+const restCLIInput = {
   name: 'productReview',
+  controllerType: ControllerGenerator.REST,
+  id: 'number',
 };
-const withInputName = testUtils.givenAControllerPath(
-  null,
-  'product-review.controller.ts',
+
+// Expected File Name
+const expectedFile = path.join(
+  SANDBOX_PATH,
+  '/src/controllers/product-review.controller.ts',
 );
 
 describe('controller-generator extending BaseGenerator', baseTests);
 describe('generator-loopback4:controller', tests);
 
 describe('lb4 controller', () => {
+  beforeEach('reset sandbox', () => sandbox.reset());
+
   it('does not run without package.json', () => {
-    return testUtils
-      .runGeneratorWith(
-        generator,
-        withInputProps,
-        testUtils.givenAnApplicationDir,
-      )
-      .then(() => {
-        assert.noFile(withInputName);
-      });
+    return expect(
+      testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () =>
+          testUtils.createLB4Project(SANDBOX_PATH, {excludePackageJSON: true}),
+        )
+        .withPrompts(basicCLIInput),
+    ).to.be.rejectedWith(/No package.json found in/);
   });
 
   it('does not run without the loopback keyword', () => {
-    return testUtils
-      .runGeneratorWith(
-        generator,
-        withInputProps,
-        testUtils.givenAnApplicationDir,
-      )
-      .then(() => {
-        assert.noFile(withInputName);
-      });
+    return expect(
+      testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () =>
+          testUtils.createLB4Project(SANDBOX_PATH, {excludeKeyword: true}),
+        )
+        .withPrompts(basicCLIInput),
+    ).to.be.rejectedWith(/No `loopback` keyword found in/);
   });
 
-  describe('basic', () => {
-    describe('with input', () => {
-      let tmpDir;
-      before(() => {
-        return testUtils
-          .runGeneratorWith(generator, withInputProps, dir => {
-            tmpDir = dir;
-            testUtils.givenAnApplicationDir(dir);
-          })
-          .toPromise();
-      });
-
-      it('writes correct file name', () => {
-        assert.file(tmpDir + withInputName);
-        assert.noFile(tmpDir + templateName);
-      });
-
-      it('scaffolds correct files', () => {
-        checkBasicContents(tmpDir);
-      });
+  describe('basic controller', () => {
+    it('scaffolds correct file with input', () => {
+      return testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () => testUtils.createLB4Project(SANDBOX_PATH))
+        .withPrompts(basicCLIInput)
+        .then(() => {
```

Comment:
Have you considering using `await` instead of `.then()` promise chaining?

In the past, we could not use async/await in plan JS files because of Node.js 6.x. Now that we don't support 6.x anymore, I think we should leverage async/await everywhere it makes sense.

```js
await testUtils
  .executeGenerator(generator)
  .inDir(...)
  .withPrompts(...);

assert.file(expectedFile);
// ...
```

Reply:
I had considered it and was running into some issues -- so I was just trying to get the tests working with promises. Issue was unrelated and I forgot to change it to async/await after. 

Will change.